microdnf/context.cpp

```diff
@@ -78,26 +78,32 @@ namespace {
 
 class MicrodnfRepoCB : public libdnf::repo::RepoCallbacks {
 public:
-    explicit MicrodnfRepoCB(libdnf::ConfigMain & config) : config(&config) {}
+    explicit MicrodnfRepoCB(libdnf::ConfigMain & config, bool quiet) : config(&config), quiet(quiet) {}
 
     void start(const char * what) override {
-        progress_bar.set_description(what);
-        progress_bar.set_auto_finish(false);
-        progress_bar.set_total_ticks(0);
-        progress_bar.start();
+        if (!quiet) {
```

Comment:
Instead of all these checks, could you just not set the progress callback in quiet mode?

Reply:
The repository callback class must be set. Because it handles the import of storage keys, not just progress indicators.
But, ... I made a change. I divide the implementation of the callback class into two classes. One "quiete" implements only the import of storage key callbacks, and the other (inherited) adds progress bars.